### PR TITLE
Added __repr__ in BaseContainer class

### LINKED ISSUE #636: Consider implementing __str__ / __repr__ for suitable containers
# Bug report

## What's wrong

When writing test cases for a method that returns `Maybe[T]`, I found that pytest reports a failure as follows which is not very readable:
> <returns.maybe._Some object at 0x7fddfe56c140> != <returns.maybe._Some object at 0x7fddfe5aea00>

## How is that should be

It would be much better, if `Some` implemented `__str__` (and maybe also `__repr__`) so that it could look like below:
> <Some(10) != Some(20)>

## System information

- `python` version: 3.8
- `returns` version: 0.14.0

### PR DESCRIPTION
**Pull request by raoashish10**
# I have made things!

## Checklist

- [x] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [x] I have created at least one test case for the changes I have made
- [x] I have updated the documentation for the changes I have made
- [x] I have added my changes to the `CHANGELOG.md`

## Related issues

- Closes #636 
Added the __ repr __() to the BaseContainer class. I haven't removed the __ str __(), I  have just kept both. Let me know if any changes are required.

### CODE REVIEW

File: returns/primitives/container.py [hunk 1]
```diff
@@ -29,6 +29,13 @@ def __str__(self) -> str:
             str(self._inner_value),
         )
 
+    def __repr__(self) -> str:
```
Comment (sobolevn) [inline returns/primitives/container.py:32]: There's no need to add a new method, we can just rename `__str__` to `__repr__`

**Review by sobolevn — CHANGES_REQUESTED** [review]
Thanks a lot! 

Please, see what tests will fail and fix them as well.

File: returns/primitives/container.py (hunk 1 above)
Comment (raoashish10) [reply returns/primitives/container.py:32]: Sure I will do that!

File: returns/primitives/container.py [hunk 2]
```diff
@@ -22,7 +22,7 @@ def __init__(self, inner_value) -> None:
         """
         object.__setattr__(self, '_inner_value', inner_value)  # noqa: WPS609
 
-    def __str__(self) -> str:
+    def __repr__(self) -> str:
         """Converts to string."""
```
Comment (thepabloaguilar) [inline returns/primitives/container.py:26]: Can you update the docstring, please?
`__repr__` is more than a simple string conversion.

Comment (raoashish10) [reply returns/primitives/container.py:26]: Yes I will change it give me a minute. I had kept it as """Used to display details of the object""" before. Is that fine?

Comment (sobolevn) [reply returns/primitives/container.py:26]: 👍

**Review by sobolevn — APPROVED** [review]

**Comment by sobolevn** [comment]
Thanks a lot! 👍

**Comment by raoashish10** [comment]
@sobolevn I look to contribute to bigger issues in the future!

**Comment by sobolevn** [comment]
@raoashish10 feel free to choose any other issues from the backlog.

**Pull request merged**